### ISSUE TITLE
Allow wide union nodes in query plan to be abbreviated when `LazyFrame.show_graph()`

### ISSUE THREAD

**Issue by OneRaynyDay** [pola-rs/polars#4674]
### Problem Description

When crafting a `pl.LazyFrame` query, I look at the query plan to see what's getting pushed down, what the chain of transformations has been, and I think it helps a lot with having a global mental model of the state of the graph. Unfortunately, there are a couple cases where the graph gets too large for me to view:

1. When I `scan_parquet` thousands of files and union them
2. When I rederive a quantity from hundreds of rows (hundreds of expressions in a select)

I see we already do this truncation in the graph when we have a really long filter. Can we have the same for the above two cases?

**Comment by ritchie46** [comment]
Yes.. ideally I'd like those outputs to be more tunable. As it's hard to predict what you want to see.
